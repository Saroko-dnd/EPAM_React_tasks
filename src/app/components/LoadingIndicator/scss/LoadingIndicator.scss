$color-red: rgb(241, 15, 15);
$loading-font-color: $color-red;

.loading-indicator {
  font-size: 2rem;
  font-weight: bold;
  color: $loading-font-color;
}
