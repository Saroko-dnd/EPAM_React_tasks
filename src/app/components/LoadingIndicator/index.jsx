import React from 'react';
import { RingLoader } from 'react-spinners';
import PropTypes from 'prop-types';
import ExtraPropTypes from 'react-extra-prop-types';

import constants from '../../constants';
import './LoadingIndicator.scss';

const LoadingIndicator = ({ color, size, message }) => (
    <div className="loading-indicator d-flex flex-column align-items-center">
        <RingLoader color={color} size={size} />
        <span>{message}</span>
    </div>
);

LoadingIndicator.propTypes = {
    color: ExtraPropTypes.color,
    size: PropTypes.number,
    message: PropTypes.string,
};

LoadingIndicator.defaultProps = {
    color: constants.colors.green,
    size: 150,
    message: 'Loading...',
};

export default LoadingIndicator;
