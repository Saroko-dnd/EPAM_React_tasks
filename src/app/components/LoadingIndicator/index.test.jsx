import LoadingIndicator from '.';

describe('render a LoadingIndicator', () => {
    const loadingIndicator = mount(<LoadingIndicator />);

    it('should render LoadingIndicator correctly', () => {
        expect(loadingIndicator).toMatchSnapshot();
    });
});
