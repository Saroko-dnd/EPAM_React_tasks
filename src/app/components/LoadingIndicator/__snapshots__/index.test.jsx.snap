// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a LoadingIndicator should render LoadingIndicator correctly 1`] = `
<LoadingIndicator
  color="#009900"
  message="Loading..."
  size={150}
>
  <div
    className="loading-indicator d-flex flex-column align-items-center"
  >
    <onlyUpdateForKeys(Loader)
      color="#009900"
      loading={true}
      size={150}
      sizeUnit="px"
    >
      <Loader
        color="#009900"
        loading={true}
        size={150}
        sizeUnit="px"
      >
        <div
          className="css-d7h8wz"
        >
          <div
            className="css-1dil71f"
          />
          <div
            className="css-18gpygh"
          />
        </div>
      </Loader>
    </onlyUpdateForKeys(Loader)>
    <span>
      Loading...
    </span>
  </div>
</LoadingIndicator>
`;
